### DISCUSSION TITLE
How to use self-hosted Gitlab as an identity provider

### DISCUSSION THREAD

**Discussion by inlann** [keycloak/keycloak#23807] [Q&A] [answered]
Hello:

Glad to find this awesome framework!!

I noticed that the Keycloak can use official Gitlab or Github as the identity provider:

![image](https://github.com/keycloak/keycloak/assets/33852855/517e871a-ab0b-4df7-afcb-0308df185e36)

And I'm allowed to modify the base url of the Github settings:

![image](https://github.com/keycloak/keycloak/assets/33852855/1788143a-cd8a-46ed-b9e5-e944dcb87651)

But I cannot find the same base url input box in the Gitlab settings:

![image](https://github.com/keycloak/keycloak/assets/33852855/d375fc21-2078-45e5-acbb-0fcd0eed19ee)

How can I add a self-hosted Gitlab as an identity provider?

Really appreciate!

**Comment by inlann** [comment, marked as answer]
It's done!

**Reply by bubnovd** [reply]
@inlann  How did you do it?

**Reply by inlann** [reply]
Hi @bubnovd 

You can add your self-hosted Gitlab as an 'OpenID Connect v1.0':
<img width="1845" alt="image" src="https://github.com/keycloak/keycloak/assets/33852855/88f1ba5a-ffef-41e6-b095-d567518be5a9">
and configure it:
![image](https://github.com/keycloak/keycloak/assets/33852855/59742684-266d-43f5-82ee-08749ce57e16)

And then, add it into authentication flow.
